### ISSUE TITLE
Option to enable auto refresh

### ISSUE THREAD

**Issue by zcourts** [Canop/broot#730]
First of all, great work! I've found myself using this quite a lot.
Before my workflow included a lot of ls, tree, ncdu and other commands and I've now adopted to using broot for all of them.

One workflow I've not been able to reproduce is where I previously did `watch tree -aDht /path/to/some/{dir1, dir2, dir3}`.

I think there are two potential features in here:

1. I'm using watch because it is a file heavy workflow and I'm monitoring what's going on with them. broot has a refresh but requires either hitting F5 or typing refresh (from what I've been able to work out anyway). The feature request here is to add an option to enable auto refresh after say 2 seconds (tree command default). That allows me to glance at the other monitor to see what's happening without needing to switch keyboard focus
2. The second potential feature request is an option to expand the entire tree that is visible. I can't work out what rules broot is using to auto hide some directories. In my workflow, I need to see the entire tree. Not everything fits on screen, but that's fine. Ideally, I'll scroll into view the parts of the tree I want to be able to monitor right now and when broot auto refreshes, it keeps the same scroll position. If the tree has changed because a file was added or removed then scroll to the nearest sibling of the file or directory that has been removed. If files or directories were added, stay in the same position and anything that falls outside the view is just not visible now. In most cases I have a sub-tree anyway as shown in the example `{dir1, dir2, dir3}`. Here, I am choosing a sub-tree I expect to mostly fit on screen

**Comment by Canop** [comment]
Auto-refresh using notify, that could be a feature.

**Comment by zcourts** [comment]
Do you mean sending desktop notifications?

**Comment by Canop** [comment]
No, I mean refreshing on file changes, using a generalization of [inotify](https://man7.org/linux/man-pages/man7/inotify.7.html). This is much more efficient than doing periodical refreshes.

**Comment by zcourts** [comment]
Ah, yes I think that makes sense! My description said refresh but I'd expect something like inotify, there's a [notify crate](https://docs.rs/notify/latest/notify/index.html#examples) as well.

**Comment by Canop** [comment]
Yes, I know that (I'm the author of [bacon](https://github.com/Canop/bacon), too)

**Comment by Dimagog** [comment]
+1 on this feature request, want it badly for the work I'm currently doing

**Comment by VKondakoff** [comment]
+1 on auto-refresh.

**Comment by darkrock** [comment]
Want to add another +1 for auto-refresh

**Comment by Canop** [comment]
> Ohhhhh! Who do I test? Build locally from the `watch` branch?

Yes. If you'd like help to compile, drop by [the chat](https://miaou.dystroy.org/3490)

**Comment by zcourts** [comment]
I see my brain converted "How" to "Who"....

Anyway, I had a quick test and it is looking good so far. I will have to use it in earnest to provide more feedback.
First initial thought is that some debounce may be needed. I modified one file to test and it jumped all over the place very quickly then slowed and settled...so I presume from what I saw that it is firing more than once because I'm looking at a directory but modified a file nested 3 or four levels down. Perhaps it fired once for each folder in the path down to the file I modified?

I'll do some more testing and let you know

**Comment by Canop** [comment]
> First initial thought is that some debounce may be needed.

I'll have a look

**Comment by Canop** [comment]
inotify events are debounced:
- an isolated event sends a refresh immediately
- successive events after the first one will have to wait a little
- there's at most one refrest sent every DEBOUNCE_MAX_DELAY
- if there's a long sequence of events, it's guaranteed that there's one
   refresh sent every DEBOUNCE_MAX_DELAY
- the last event of the sequence is always sent (with a delay of
   at most DEBOUNCE_MAX_DELAY), ensuring we don't miss any change

Right now, DEBOUNCE_MAX_DELAY is 500ms
